drivers/led/issi/is31fl3761.h

```diff
@@ -0,0 +1,610 @@
+#define IS31FL3761_I2C_ADDRESS_VCC_SDA 0x3E
+#define IS31FL3761_I2C_ADDRESS_VCC_VCC 0x3F
+
+#if defined(RGB_MATRIX_IS31FL3761)
+#    define IS31FL3761_LED_COUNT RGB_MATRIX_LED_COUNT
+#endif
+
+#if defined(IS31FL3761_I2C_ADDRESS_4)
+#    define IS31FL3761_DRIVER_COUNT 4
+#elif defined(IS31FL3761_I2C_ADDRESS_3)
+#    define IS31FL3761_DRIVER_COUNT 3
+#elif defined(IS31FL3761_I2C_ADDRESS_2)
+#    define IS31FL3761_DRIVER_COUNT 2
+#elif defined(IS31FL3761_I2C_ADDRESS_1)
+#    define IS31FL3761_DRIVER_COUNT 1
+#endif
+
+typedef struct is31fl3761_led_t {
+    uint8_t  driver : 2;
+    uint16_t r : 10;
+    uint16_t g : 10;
+    uint16_t b : 10;
+} PACKED is31fl3761_led_t;
+
+extern const is31fl3761_led_t PROGMEM g_is31fl3761_leds[IS31FL3761_LED_COUNT];
+
+void is31fl3761_init_drivers(void);
+void is31fl3761_init(uint8_t index);
+void is31fl3761_write_register(uint8_t index, uint8_t reg, uint8_t data);
+void is31fl3761_select_page(uint8_t index, uint8_t page);
+
+void is31fl3761_set_color(int index, uint8_t red, uint8_t green, uint8_t blue);
+void is31fl3761_set_color_all(uint8_t red, uint8_t green, uint8_t blue);
+
+void is31fl3761_set_led_control_register(uint8_t index, bool red, bool green, bool blue);
+
+// This should not be called from an interrupt
+// (eg. from a timer interrupt).
+// Call this while idle (in between matrix scans).
+// If the buffer is dirty, it will update the driver with the buffer.
+void is31fl3761_update_pwm_buffers(uint8_t index);
+void is31fl3761_update_led_control_registers(uint8_t index);
+void is31fl3761_set_scaling_registers(const is31fl3761_led_t *pled, uint8_t red, uint8_t green, uint8_t blue);
+
+void is31fl3761_set_pwm_buffer(const is31fl3761_led_t *pled, uint8_t red, uint8_t green, uint8_t blue);
+
+void is31fl3761_flush(void);
+
+// SWPDR SWx Pull down Resister Selection Bit
+#define IS31FL3761_PDR_0_OHM 0b000   // No pull up
+#define IS31FL3761_PDR_0K5_OHM 0b001 // 2.6V
+#define IS31FL3761_PDR_1K_OHM 0b010  // 2.2V
+#define IS31FL3761_PDR_2K_OHM 0b011  // 1.9V
+#define IS31FL3761_PDR_4K_OHM 0b100  // 1.6V
+#define IS31FL3761_PDR_8K_OHM 0b101  // 1.2V
+#define IS31FL3761_PDR_16K_OHM 0b110 // 0.9V
+#define IS31FL3761_PDR_32K_OHM 0b111 // GND
+
+// CSPUR  CSy Pull up Resister Selection Bit
+#define IS31FL3761_PUR_0_OHM 0b000   // No pull up
+#define IS31FL3761_PUR_0K5_OHM 0b001 // PVCC-2.3V
+#define IS31FL3761_PUR_1K_OHM 0b010  // PVCC-2.0V
+#define IS31FL3761_PUR_2K_OHM 0b011  // PVCC-1.7V
+#define IS31FL3761_PUR_4K_OHM 0b100  // PVCC-1.4V
+#define IS31FL3761_PUR_8K_OHM 0b101  // PVCC-1.1V
+#define IS31FL3761_PUR_16K_OHM 0b110 // PVCC-0.8V
+#define IS31FL3761_PUR_32K_OHM 0b111 // PVCC
+
+#define IS31FL3761_PWM_FREQUENCY_92K_HZ 0b000  // 92kHz for 8-bit mode
+#define IS31FL3761_PWM_FREQUENCY_46k_HZ 0b001  // 46kHz for 8-bit mode
+#define IS31FL3761_PWM_FREQUENCY_23k_HZ 0b110  // 23kHz for 8-bit mode
+#define IS31FL3761_PWM_FREQUENCY_11k5_HZ 0b011 // 11.5kHz for 8-bit mode
```

Comment:
These names should be fixed to match the datasheet better (for the PWM frequency you may choose to match the 6+2-bit mode, which seems to be the default in your code).